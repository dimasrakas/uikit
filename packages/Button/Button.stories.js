import Button from "./Button";

export default {
  title: "Design System/Button",
  component: Button,
};

// More on component templates: https://storybook.js.org/docs/vue/writing-stories/introduction#using-args
const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { Button },
  template: '<Button v-bind="$props" />',
});

export const Primary = Template.bind({});
// More on args: https://storybook.js.org/docs/vue/writing-stories/args
Primary.args = {
  label: "Button",
};
