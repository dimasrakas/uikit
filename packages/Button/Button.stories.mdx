import { Meta, Story, Canvas } from "@storybook/addon-docs/blocks";
import { ArgsTable } from "@storybook/addon-docs";

import Button from "./Button.vue";

<Meta
  title="Design System/Button"
  component={Button}
  argTypes={{
    size: { control: "select", options: ["default", "sm"] },
    variant: { control: "radio", options: ["primary", "secondary"] },
  }}
/>

export const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { Button },
  template: '<Button v-bind="$props" />',
});

# Button

With `MDX`, we can define a story for `Button` right in the middle of our
Markdown documentation.

## Install

```bash
yarn add @dimasrakas/button

// or

npm install @dimasrakas/button
```

## Imports

```js
import { Button } from "@dimasrakas/button";
```

## Usage

<Canvas withSource="open">
  <Story
    name="Default"
    args={{
      label: "Default Button",
      variant: "primary",
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

## Props

<ArgsTable of={Button} />
