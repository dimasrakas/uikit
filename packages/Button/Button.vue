<template>
  <button>{{ label }}</button>
</template>

<script>
export default {
  name: "Button",
  props: {
    label: {
      type: String,
      default: "Button",
    },
  },
};
</script>

<style lang="scss" scoped></style>
