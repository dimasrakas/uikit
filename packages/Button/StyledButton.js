import styled from "vue-styled-components";
import { space, color } from "styled-system";

const StyledButton = styled("button", {
  mt: [Number, String, Array],
  px: [Number, String, Array],
  py: [Number, String, Array],
  bg: String,
  color: String,
})`
  border: none;
  border-radius: 0.5rem;
  font-family: ui-sans-serif, system-ui, -apple-system, BlinkMacSystemFont,
    "Segoe UI", Roboto, "Helvetica Neue", Arial, "Noto Sans", sans-serif,
    "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji";
  font-size: 100%;
  line-height: inherit;
  color: white;
  margin: 0;
  padding: 0;
  cursor: pointer;
  &:hover {
    opacity: 0.9;
  }

  ${space}
  ${color}
`;

export default StyledButton;
