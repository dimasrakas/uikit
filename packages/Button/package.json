{
  "name": "@dimasrakas/button",
  "version": "1.0.0",
  "publishConfig": {
    "access": "public"
  },
  "dependencies": {
    "styled-system": "4.2.4",
    "vue-styled-components": "1.6.0"
  }
}
