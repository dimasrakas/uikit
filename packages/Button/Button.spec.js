import { shallowMount } from "@vue/test-utils";
import Button from "./Button.vue";

describe("Button.vue", () => {
  it('should have "Button" as name', () => {
    expect(Button.name).toBe("Button");
  });

  it("should render correctly", () => {
    const wrapper = shallowMount(Button);
    expect(wrapper.element).toMatchSnapshot();
  });
});
