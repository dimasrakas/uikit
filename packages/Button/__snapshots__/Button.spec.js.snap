// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.vue should render correctly 1`] = `
<button>
  Button
</button>
`;
